crash interpolating bytes with dictionary

```
$ cat example.py
print(b'%(x)s' % {b'x': b'data'})
$ python3 example.py
b'data'
$ mypy example.py --show-traceback
mypy example.py  --show-traceback
example.py:1: error: INTERNAL ERROR -- Please try using mypy master on Github:
https://mypy.rtfd.io/en/latest/common_issues.html#using-a-development-mypy-build
Please report a bug at https://github.com/python/mypy/issues
version: 0.720
Traceback (most recent call last):
  File "mypy/checkexpr.py", line 3404, in accept
  File "mypy/nodes.py", line 1708, in accept
  File "mypy/checkexpr.py", line 1891, in visit_op_expr__ExpressionVisitor_glue
  File "mypy/checkexpr.py", line 1906, in visit_op_expr
  File "mypy/checkstrformat.py", line 97, in check_str_interpolation
  File "mypy/checkstrformat.py", line 178, in check_mapping_str_interpolation
TypeError: StrExpr object expected
example.py:1: : note: use --pdb to drop into pdb
```

--- a/mypy/checkstrformat.py
+++ b/mypy/checkstrformat.py
@@ -174,11 +174,11 @@ class StringFormatterChecker:
                                         replacements: Expression,
                                         expr: FormatStringExpr) -> None:
         if (isinstance(replacements, DictExpr) and
-                all(isinstance(k, (StrExpr, BytesExpr))
+                all(isinstance(k, (StrExpr, BytesExpr, UnicodeExpr))
                     for k, v in replacements.items)):
             mapping = {}  # type: Dict[str, Type]
             for k, v in replacements.items:
-                key_str = cast(StrExpr, k).value
+                key_str = cast(FormatStringExpr, k).value
                 mapping[key_str] = self.accept(v)
 
             for specifier in specifiers:

--- a/test-data/unit/check-expressions.test
+++ b/test-data/unit/check-expressions.test
@@ -1223,6 +1223,7 @@ b'%a' % 3
 [case testStringInterpolationMappingTypes]
 '%(a)d %(b)s' % {'a': 1, 'b': 's'}
 '%(a)d %(b)s' % {'a': 's', 'b': 1}  # E: Incompatible types in string interpolation (expression has type "str", placeholder with key 'a' has type "Union[int, float]")
+b'%(x)s' % {b'x': b'data'}
 [builtins fixtures/primitives.pyi]
 
 [case testStringInterpolationMappingKeys]

